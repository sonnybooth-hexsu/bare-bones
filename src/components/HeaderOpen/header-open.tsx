import React, { useState } from "react"
import styles from "./header-open.module.css"
import { X, ChevronDown, ChevronUp } from "react-feather"

type HeaderOpenProps = {
  siteTitle: string
  navLinks: Array<{
    name: string
    page: string
    id: number
    subLinks: Array<{ name: string; page: string; id: number }>
  }>
  navToggle: () => Function
}

const SubMenuItem = ({ name, subLinks, id }) => {
  const [openState, setOpenState] = useState(false)

  return (
    <li
      key={`navigation-item-${id}`}
      className={`animate-reveal delay-${id}`}
      onClick={() => setOpenState(!openState)}
    >
      <span className={styles.headerNavigationSubMenu}>
        {name}
        {openState ? <ChevronUp /> : <ChevronDown />}
      </span>
      <div className={`${openState ? "block" : "hidden"}`}>
        {subLinks.map(({ name, page, id }) => (
          <a key={`navigation-sublink-item-${id}`} href={page}>
            {name}
          </a>
        ))}
      </div>
    </li>
  )
}

export const HeaderOpen = ({
  siteTitle,
  navLinks,
  navToggle,
}: HeaderOpenProps) => {
  return (
    <div className={styles.headerContainer}>
      <div className="container mx-auto pl-6 pr-6 md:pl-0 md:pr-0">
        <div className={styles.header}>
          <a href="/" className={styles.headerLogo}>
            {siteTitle}
          </a>
          <X className={styles.headerMenuIcon} onClick={navToggle} />
        </div>
        <nav className={styles.headerNavigation}>
          <ol>
            {navLinks.map(({ name, page, subLinks, id }) => {
              return subLinks.length > 0 ? (
                <SubMenuItem
                  key={`navigation-item-${id}`}
                  name={name}
                  subLinks={subLinks}
                  id={id}
                />
              ) : (
                <li
                  className={`animate-reveal delay-${id}`}
                  data-testid="headerNavigationLink"
                  key={`navigation-item-${id}`}
                >
                  <a href={`${page}`}>{name}</a>
                </li>
              )
            })}
          </ol>
        </nav>
      </div>
    </div>
  )
}
