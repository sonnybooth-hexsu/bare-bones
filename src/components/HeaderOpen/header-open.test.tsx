/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { queryAllByTestId } from "@testing-library/dom"
import { HeaderOpen } from "./header-open"

const testNavLinks = [
  {
    name: `Link 1`,
    page: `link-1`,
    subLinks: [],
    id: 1,
  },
  {
    name: `Link 2`,
    page: `link-2`,
    subLinks: [],
    id: 2,
  },
  {
    name: `Link 3`,
    page: `link-3`,
    subLinks: [],
    id: 3,
  },
]

afterEach(() => {
  cleanup()
})

describe("<Header />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <HeaderOpen
        siteTitle="Test Site"
        navLinks={testNavLinks}
        navToggle={() => Function}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should have the length of links passed through", () => {
    const { container } = render(
      <HeaderOpen
        siteTitle="Test Site"
        navLinks={testNavLinks}
        navToggle={() => Function}
      />
    )

    expect(
      queryAllByTestId(container, "headerOpenNavigationLink")
    ).toHaveLength(testNavLinks.length)
  })
})
