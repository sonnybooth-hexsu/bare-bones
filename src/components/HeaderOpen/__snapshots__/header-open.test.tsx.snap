// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <div
    class="headerContainer"
  >
    <div
      class="container mx-auto pl-6 pr-6 md:pl-0 md:pr-0"
    >
      <div
        class="header"
      >
        <a
          class="headerLogo"
          href="/"
        >
          Test Site
        </a>
        <svg
          class="headerMenuIcon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </div>
      <nav
        class="headerNavigation"
      >
        <ol>
          <li
            class="animate-reveal delay-1"
            data-testid="headerOpenNavigationLink"
          >
            <a
              href="link-1"
            >
              Link 1
            </a>
          </li>
          <li
            class="animate-reveal delay-2"
            data-testid="headerOpenNavigationLink"
          >
            <a
              href="link-2"
            >
              Link 2
            </a>
          </li>
          <li
            class="animate-reveal delay-3"
            data-testid="headerOpenNavigationLink"
          >
            <a
              href="link-3"
            >
              Link 3
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
</div>
`;
